src/main/java/sim/coffee/Kitchen.java

```diff
@@ -0,0 +1,70 @@
+package sim.coffee;
+
+import java.util.LinkedList;
+
+public class Kitchen implements Runnable, Subject {
+	private SharedQueue kitchenQueue;
+    private StringBuilder currentOrder;
+    private LinkedList<Observer> observers;
+	private Logger log = Logger.getInstance();;
+	
+	public Kitchen(SharedQueue kitchenQueue) {
+		this.kitchenQueue = kitchenQueue;
+		currentOrder = new StringBuilder();
+		observers = new LinkedList<>(); 
+	}
+
+	@Override
+	public void run() {
+
+        // Service continues as long as customers are still due to arrive or customers
+        // are in the queue
+        while (!kitchenQueue.getDone() || !kitchenQueue.isEmpty()) {
+            Order [] order = kitchenQueue.getCustomerOrder();
+            setCurrentOrder(order);
+            notifyObservers();
+        	try {
+                // Time to process order depends on number of items
+                Thread.sleep(5000l * order.length);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        	log.add(order, Logger.OrderState.SERVED);
+        }
+        currentOrder.replace(0, currentOrder.length(), "");
+        notifyObservers();
+        log.writeReport("log.txt");
+    }
+		
+    // adds details of the order being processed by the Kitchen
+    public void setCurrentOrder(Order[] o) {
+    	currentOrder.replace(0, currentOrder.length(), "");
```

Comment:
I'm not sure how I feel about all this sting builder stuff in the model 🤔 

It doesn't feel super cohesive to me because it's like the subject/model is preparing the data for a specific observer's use case (i.e. it almost feels like the view using just the raw data for this string building stuff in its update method).

Reply:
That makes sense, would make the class more reusable. I've moved the StringBuilder from Server and Kitchen classes to update() in the GUI classes now.